cannon/mipsevm/memory/hasher.go

```diff
@@ -0,0 +1,74 @@
+package memory
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Byte32Pool is a sync.Pool for [32]byte slices
+var Byte32Pool = sync.Pool{
```

Comment:
(nit) Some of these variables / symbols are not used externally.  Might be a bit clearer if we lower-case the internal symbols. 
```suggestion
var byte32Pool = sync.Pool{
```